### PR TITLE
add new steam dirs for Unity IWAD and Final Doom

### PR DESCRIPTION
**Pull request by rfomin**
Taken from GZDoom: https://github.com/ZDoom/gzdoom/blob/0eae1b9be25dd242ee3cb9d5ced519de2af832b8/src/win32/i_steam.cpp#L224

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -551,14 +551,10 @@ void D_DoAdvanceDemo(void)
   name = demostates[demosequence][gamemode].name;
   if (name && !strcasecmp(name, "TITLEPIC"))
   {
-    int i = W_CheckNumForName("TITLEPIC");
-    int j = W_CheckNumForName("DMENUPIC");
-
-    if (i < 0 || (j >= 0 && W_IsIWADLump(i)))
-      name = (j >= 0) ? "DMENUPIC" : "INTERPIC";
+    if (W_CheckNumForName("TITLEPIC") < 0)
+      name = "DMENUPIC";
```
Comment (rfomin) [inline src/d_main.c:555]: I didn't understand these checks.
Unity IWAD has both `TITLEPIC` and `DMENUPIC`.

Comment (fabiangreffrath) [reply src/d_main.c:555]: The Doom 2 IWAD that comes with the Doom 3 BFG Edition is missing the TITLEPIC lump and is supposed to show DMENUPIC instead.

Comment (rfomin) [reply src/d_main.c:555]: Yes, but why is there `W_IsIWADLump(i)`? Do my changes work too?

Comment (fabiangreffrath) [reply src/d_main.c:555]: If the IWAD provided both lumps, I wanted DMENUPIC to take precedence for some reason. 🤔

Comment (rfomin) [reply src/d_main.c:555]: Unity IWAD has both, but only TITLEPIC is wide.

Comment (fabiangreffrath) [reply src/d_main.c:555]: Ah, now that's a reason to prefer that instead!

**Pull request merged**